controllers/machinedeletionremediation_controller.go

```diff
@@ -72,28 +72,30 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 
 	log.Info("reconciling...")
 
-	//fetch the remediation
+	var err error
 	var remediation *v1alpha1.MachineDeletionRemediation
-	if remediation = r.getRemediation(ctx, req); remediation == nil {
-		return ctrl.Result{}, nil
+	if remediation, err = r.getRemediation(ctx, req); remediation == nil || err != nil {
+		return ctrl.Result{}, err
 	}
 
-	var machine *unstructured.Unstructured
-	//Health check was done by NHC
-	if node, err := r.getNodeFromMdr(remediation); err == nil {
-		if machine, err = r.buildMachineFromNode(node); err != nil {
-			r.Log.Error(err, "failed to fetch machine of node", "node name", node.Name)
-			return ctrl.Result{}, err
-		}
-		if !hasControllerOwner(machine) {
-			r.Log.Info("ignoring remediation of machine associated to node, since the machine has no controller owner", "node name", remediation.Name)
-			return ctrl.Result{}, nil
-		}
+	var node *v1.Node
+	if node, err = r.getNodeFromMdr(remediation); err != nil {
+		return ctrl.Result{}, err
+	}
 
-	} else { //Failed fetching the node
-		r.Log.Error(err, "failed to fetch node", "node name", remediation.Name)
+	log.Info("MDR CR and affected Node found", "node", node.GetName())
+
+	var machine *unstructured.Unstructured
+	if machine, err = r.buildMachineFromNode(node); err != nil {
+		r.Log.Error(err, "failed to fetch machine of node", "node name", node.Name)
```

Comment:
I don't think `a` is correct here. We're not looking for a generic node, but a specific one. In fact, once printed, the message looks like the following

`failed to fetch the machine of node <node name>`

IMHO logs are usually short informative sentences, not necessarily complete phrases (also "failed" would technically be incorrect, since there is no subject).